### PR TITLE
next.JS middleware

### PR DESCRIPTION
**Pull request by HonduQ**
***** `NEW ENVIRONMENT VARIABLE -- BREAKING CHANGE` *****

This pull request is to add `middleware authentication` features between the `django`, `next.js` and `react` services.
I will do this by: 
1) adding `next.js middleware`
2) creating a `SECRET_MIDDLE_KEY` in the `Django .env` and `NextJS .env` files (and do the same for React, I think)
3) verifying each secret key any time a `request` or `response` is made, thereby preventing `requests coming from an outside source`

Later: Will need to update the AWS EC-2 server and provide legit secret keys.

### CODE REVIEW

**Comment by HonduQ** [comment]
`Django --> NestJS` authentication is good, still need to work out a `rejection clause` for `bad secret keys`.

File: haec-user-service/src/_middleware.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { Injectable, NestMiddleware, Request, Response } from '@nestjs/common';
+import { NextFunction } from 'express';
+
+@Injectable()
+export class AuthenticationMiddleware implements NestMiddleware {
+
+  use(req: Request, res: Response, next: NextFunction) {
+    console.log("Middleware Authentication is running");
+    // Middleware Authentication SUCCESS
+    if (req.headers["haec-auth-token"] === "neither-is-this-a-secret-key") {
```
Comment (HonduQ) [inline haec-user-service/src/_middleware.ts:10]: I plan to update this line once this is pushed to the AWS server; is there something I should do differently now?

Comment (HonduQ) [inline haec-user-service/src/_middleware.ts:10]: I plan to update this line once this is pushed to the AWS server; is there something I should do differently now?

File: haec-user-service/src/app.module.ts [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
+import { UserService } from './modules/user/services/user.service';
+import { UserController } from './modules/user/controllers/user.controller';
 import { PrismaService } from './providers/prisma.service';
 import { ConfigModule } from '@nestjs/config';
+import { AuthenticationMiddleware } from './_middleware';
+
 
 @Module({
-  imports: [UserModule, ConfigModule.forRoot({ isGlobal: true, cache: true })],
+  imports: [UserModule, ConfigModule.forRoot({ isGlobal: true, cache: true }), UserModule],
```
Comment (HonduQ) [inline haec-user-service/src/app.module.ts:11]: Is there a reason we have `UserModule` imported `twice`?

Comment (HonduQ) [inline haec-user-service/src/app.module.ts:11]: Is there a reason we have `UserModule` imported `twice`?

File: haec-user-service/src/modules/user/controllers/user.controller.ts [hunk 3]
```diff
@@ -63,9 +63,7 @@ export class UserController {
 
   @Post('/user/verify')
   @Header('content-type', 'application/json')
-  async verifyUser(
-    @Body() verifyUserDto: VerifyUserDto,
-  ): Promise<boolean> {
+  async verifyUser(@Body() verifyUserDto: VerifyUserDto): Promise<boolean> {
```
Comment (HonduQ) [inline haec-user-service/src/modules/user/controllers/user.controller.ts:66]: I think `this and the following changes are just syntax`, I kept getting `End of Line errors` and I wasn't able to resolve them.

Comment (HonduQ) [inline haec-user-service/src/modules/user/controllers/user.controller.ts:66]: I think `this and the following changes are just syntax`, I kept getting `End of Line errors` and I wasn't able to resolve them.

File: api/holiday_acres_api/views.py [hunk 4]
```diff
@@ -21,6 +21,7 @@
 env = environ.Env()
 environ.Env.read_env(os.path.join(BASE_DIR, ".env"))
 user_service_var = env("USER_SERVICE")
+django_secret_key = env("SECRET_MIDDLE_KEY")
```
Comment (mstephenhunt) [inline api/holiday_acres_api/views.py:24]: Can you put `SECRET_MIDDLE_KEY` in the Django `.env.example`? https://github.com/mstephenhunt/holiday_acres/blob/0ac9d6befc4757aff7c9734cd4bb370eb6c093c4/api/.env.example

File: haec-user-service/src/_middleware.ts (hunk 1 above)
Comment (mstephenhunt) [reply haec-user-service/src/_middleware.ts:10]: This should be an environment variable just like what you have in the Django service, but for the NestJS service. 

If you could provide a dummy value [here in the NestJS `.env.example` file](https://github.com/mstephenhunt/holiday_acres/blob/main/haec-user-service/.env.example), you can simply reference environment variables with a line like:

```ts
env("MY_ENV_VAR_NAME")
```

File: haec-user-service/src/app.module.ts (hunk 2 above)
Comment (mstephenhunt) [reply haec-user-service/src/app.module.ts:11]: No, this should only be imported once.

File: haec-user-service/src/modules/user/controllers/user.controller.ts (hunk 3 above)
Comment (mstephenhunt) [reply haec-user-service/src/modules/user/controllers/user.controller.ts:66]: Yeah this is the linter just going wild. Looks good to me 👍

Comment (mstephenhunt) [reply haec-user-service/src/modules/user/controllers/user.controller.ts:66]: I really gotta fix this 😬

File: haec-user-service/src/_middleware.ts (hunk 1 above)
Comment (HonduQ) [reply haec-user-service/src/_middleware.ts:10]: That makes total sense. On it.

File: haec-user-service/src/modules/user/controllers/user.controller.ts (hunk 3 above)
Comment (HonduQ) [reply haec-user-service/src/modules/user/controllers/user.controller.ts:66]: Lol

File: api/holiday_acres_api/views.py (hunk 4 above)
Comment (HonduQ) [reply api/holiday_acres_api/views.py:24]: Will do

**Pull request merged**